src/bucket/BucketListBase.h

```diff
@@ -363,6 +363,7 @@ template <class BucketT> class BucketLevel
     FutureBucket<BucketT> mNextCurr;
     std::shared_ptr<BucketT> mCurr;
     std::shared_ptr<BucketT> mSnap;
+    std::shared_ptr<BucketT> mNextCurrInMemory;
```

Comment:
Could we use std::variant to represent "next curr" in a single variable? (it'd be safer and less error-prone)

Reply:
I tried to implement this via with `std::monostate` too, but unfortunately we use a single long-lasting `FutureBucket` object per level and reset it manually. Many callers assume this object exists, and it was too big of a refactor to implement initializing the FutureBucket correctly wrt monostate.